### PR TITLE
Form loading

### PR DESCRIPTION
**Pull request by ncphillips**

### CODE REVIEW

**Comment by dwalkr** [comment]
#1219

File: packages/@tinacms/forms/src/form.test.ts [hunk 1]
```diff
@@ -36,6 +37,67 @@ describe('Form', () => {
...
+      })
+      describe('after promise resolves', () => {
+        it('is not loading once resolved', async () => {
+          let resolve: () => void | null
+
+          const form = new Form({
+            ...DEFAULTS,
+            loadInitialValues: () =>
+              new Promise(r => {
+                resolve = r
```
Comment (dwalkr) [inline packages/@tinacms/forms/src/form.test.ts:74]: :exploding_head:

Comment (ncphillips) [reply packages/@tinacms/forms/src/form.test.ts:74]: Shocked by the crazy, resolve a promise outside of the promise shenanigans?

Comment (dwalkr) [reply packages/@tinacms/forms/src/form.test.ts:74]: I just never would have thought of it. clever

**Pull request merged**